ipatests/test_integration/test_otp.py

```diff
@@ -392,6 +456,108 @@ def test_2fa_disable_single_prompt(self):
+        master.run_command(['ipa', 'user-mod', USER3, '--user-auth-type=otp',
+                            '--user-auth-type=password'])
+        try:
+            otpuid, totp = add_otptoken(master, USER3, otptype='totp')
+            master.run_command(['ipa', 'otptoken-show', otpuid])
+            totp.generate(int(time.time())).decode('ascii')
+            otpvalue = "\n"
+            tasks.clear_sssd_cache(self.master)
+            sssd_version = tasks.get_sssd_version(master)
+            with xfail_context(sssd_version < tasks.parse_version('2.9.5'),
+                               'https://github.com/SSSD/sssd/pull/7500'):
+                result = ssh_2fa_with_cmd(master, self.master.external_hostname,
+                                        USER3, PASSWORD, otpvalue=otpvalue,
+                                        command="klist")
+                print(result.stdout_text)
+                assert 'Password authentication successful' in result.stdout_text
+                assert USER3 in result.stdout_text
+                assert (f'Default principal: {USER3}@{self.master.domain.realm}' in
+                        result.stdout_text)
+                cmd = self.master.run_command(['semanage', 'login', '-l'])
+                assert USER3 in cmd.stdout_text
+        finally:
+            master.run_command(['ipa', 'user-del', USER3])
+            self.master.run_command(['semanage', 'login', '-D'])
+            sssd_conf_backup.restore()
+
+    def test_2fa_with_otp_password(self):
+        """Test ssh with 2FA only with password and otpvalue when
+        user-auth-type is opt and password.
+
+        Test for :  https://github.com/SSSD/sssd/pull/7500
+
+        Add the IPA user and user-auth-type set to opt and password.
+        Authenticate the user only with password and otpvalue.
+        """
+        master = self.master
+        USER4 = 'sshuser4'
+        sssd_conf_backup = tasks.FileBackup(master, paths.SSSD_CONF)
+        first_prompt = 'Enter first factor:'
+        second_prompt = 'Enter second factor:'
+        add_contents = textwrap.dedent('''
+            [prompting/2fa/sshd]
+            single_prompt = False
+            first_prompt = {0}
+            second_prompt = {1}
+            ''').format(first_prompt, second_prompt)
+        set_sssd_conf(master, add_contents)
+        tasks.create_active_user(master, USER4, PASSWORD)
+        tasks.kinit_admin(master)
+
+        master.run_command(['ipa', 'user-mod', USER4, '--user-auth-type=otp',
+                            '--user-auth-type=password'])
+        try:
+            otpuid, totp = add_otptoken(master, USER4, otptype='totp')
+            master.run_command(['ipa', 'otptoken-show', otpuid])
+            otpvalue = totp.generate(int(time.time())).decode('ascii')
+            tasks.clear_sssd_cache(self.master)
+            sssd_version = tasks.get_sssd_version(master)
+            with xfail_context(sssd_version < tasks.parse_version('2.9.5'),
```

Comment:
When OTP and password are both provided, the ssh authentication is successful (the bug was happening when only the password is provided). You can remove the with xfail_context for this test.